crates/datafusion_ext/src/planner/relation/mod.rs

```diff
@@ -261,10 +268,45 @@ fn infer_func_for_file(path: &str) -> Result<OwnedTableReference> {
             schema: "public".into(),
             table: "read_bson".into(),
         },
+        "gz" => {
+            //handling compressed files with .gz extension
+            infer_func_from_compressed_file(filename)?
+        }
         ext => {
             return Err(DataFusionError::Plan(format!(
                 "unable to infer how to handle file extension: {ext}"
             )))
         }
     })
 }
+
+fn infer_func_from_compressed_file(filename: &str) -> Result<OwnedTableReference> {
+    if filename.contains(".json.gz")
+        | filename.contains(".json1.gz")
+        | filename.contains(".ndjson.gz")
```

Comment:
given that we now have this `read_json` function for non-line delimited json does it make sense to use ndjson for `.json.gz` file endings? I think it's easier to add `json.gz` in than it is to take it out, so maybe we should go with that. 

also I'm not sure why `json1.gz` is covered here, I'd remove it unless there's a reason.